language/extensions/move-table-extension/sources/Table.move

```diff
@@ -0,0 +1,44 @@
+/// Type of large-scale storage tables.
+module Extensions::Table {
+    struct Table<phantom K, phantom V> has store {
+        handle: u128
```

Comment:
Could we provide a count of contents so that modules can delete them if they are empty and would like to do so? My exploratory nft work makes use of this concept.

Reply:
+1 exposing a count might be helpful to know when it is safe to call `destroy`

Reply:
Supporting length turns some insertions (those creating new entries) and removals writing an additional metadata node in the storage. Do we consider a version of map that doesn't know its length and not safely droppable (and makes its embodying structure not droppable as well)? Because the deletion refund won't matter much, I suspect there won't be anybody who bothers to clean up tables anyway, especially the empty table itself.

And the length-aware version can be implemented on top of that.